modules/programs/piston-cli.nix

```
@@ -0,0 +1,50 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.piston-cli;
+  yamlFormat = pkgs.formats.yaml { };
+in {
+  meta.maintainers = with maintainers; [ ethancedwards8 ];
+
+  options.programs.piston-cli = {
+    enable = mkEnableOption "piston-cli, code runner";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.piston-cli;
+      defaultText = literalExample "pkgs.piston-cli";
+      description = "The piston-cli package to use.";
+    };
+
+    settings = mkOption {
+      type = yamlFormat.type;
+      default = { };
+      example = literalExample ''
+        {
+          # Code highlihting theme
+          theme = "emacs";
+          # Used for displaying your resultant output you get from the api
+          box_style = "MINIMAL_DOUBLE_HEAD";
+          # This is what is used for continuing your prompt when you go a new line
+          prompt_continuation = "...";
+          # This is the start symbol used when you are shell
+          prompt_start = ">>>";
```

```suggestion
          theme = "emacs";
          box_style = "MINIMAL_DOUBLE_HEAD";
          prompt_continuation = "...";
          prompt_start = ">>>";
```

No need for the comments :smile: